go-controller/pkg/config/cni.go

```diff
@@ -66,10 +70,35 @@ func WriteCNIConfig() error {
 	return os.Rename(f.Name(), confFile)
 }
 
+// ParseNetConf parses config in NAD spec
+func ParseNetConf(bytes []byte) (*ovncnitypes.NetConf, error) {
+	netconf := &ovncnitypes.NetConf{MTU: Default.MTU, Topology: ovntypes.Layer3AttachDefTopoType}
+	err := json.Unmarshal(bytes, &netconf)
+	if err != nil {
+		return nil, err
+	}
+	if !netconf.IsSecondary {
+		netconf.Name = ovntypes.DefaultNetworkName
+	} else {
+		// skip non-OVN nad
+		if netconf.Type != "ovn-k8s-cni-overlay" {
+			return nil, ErrorAttachDefNotOvnManaged
+		}
+		if netconf.NadName == "" {
+			return nil, fmt.Errorf("missing NadName in secondary network netconf %s", netconf.Name)
+		}
+		if netconf.Name == "" || netconf.Name == ovntypes.DefaultNetworkName {
+			return nil, fmt.Errorf("invalid name in in secondary network netconf (%s)", netconf.Name)
+		}
+	}
```

Comment:
hmm, in our deployment, nad of the default network has "ovn-primary" as its name.

Reply:
Hmm, in `WriteCNIConfig` we are using `ovn-kubernetes` and I have seen downstream tests that specify no name. We should discuss if we can rely on the NetConf name as a network name and if not it might be a beter way forward to have an extra parameter instead of using NetConf name or `IsSecondary`.